src/bilder/images/airbyte/deploy.py

```diff
@@ -0,0 +1,243 @@
+    mode="0664",
+)
+
+env_template_file = Path("/etc/consul-template/.env.tmpl")
+nginx_pomerium_conf_template_file = Path(
+    str(pomerium_config.configuration_template_directory) + "/nginx_pomerium.conf.tmpl"
+)
+nginx_pomerium_conf_file = Path(
+    str(pomerium_config.configuration_directory) + "/nginx_pomerium.conf"
+)
+nginx_htpasswd_template_file = Path(
+    str(pomerium_config.configuration_template_directory) + "/nginx_htpasswd.tmpl"
+)
+nginx_htpasswd_file = Path(
+    str(pomerium_config.configuration_directory) + "/nginx_htpasswd"
+)
+nginx_proxy_conf_file = Path(
+    str(pomerium_config.configuration_directory) + "/nginx_proxy.conf"
+)
+
+files.put(
+    name="Create the .env template file in docker-compose directory.",
+    src=str(Path(__file__).resolve().parent.joinpath("templates", ".env.tmpl")),
+    dest=str(env_template_file),
+    mode="0664",
+)
+
+files.put(
+    name="Create the pomerium nginx configuration file template",
+    src=str(
+        Path(__file__).resolve().parent.joinpath("files", "nginx_pomerium.conf.tmpl")
+    ),
+    dest=str(nginx_pomerium_conf_template_file),
+    mode="0664",
+)
+
+files.put(
+    name="Create the nginx htpasswd file template for nginx basic auth on the dagster bypass.",
+    src=str(Path(__file__).resolve().parent.joinpath("files", "nginx_htpasswd.tmpl")),
+    dest=str(nginx_htpasswd_template_file),
+    mode="0664",
+)
+
+files.put(
+    name="Create the the proxy nginx configuration file in the pomerium etc directory.",
+    src=str(Path(__file__).resolve().parent.joinpath("files", "nginx_proxy.conf")),
+    dest=str(nginx_proxy_conf_file),
+    mode="0664",
+)
+
+
+vault_templates = [
+    VaultTemplate(
+        source=pomerium_config.configuration_template_file,
+        destination=Path(pomerium_config.configuration_file),
+    ),
+    VaultTemplate(
+        source=nginx_pomerium_conf_template_file,
+        destination=nginx_pomerium_conf_file,
+    ),
+    VaultTemplate(
+        source=nginx_htpasswd_template_file,
+        destination=nginx_htpasswd_file,
+    ),
+    VaultTemplate(
+        contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+        "{{ printf .Data.key }}{{ end }}",
+        destination=Path(f"{pomerium_config.certificate_key_file}"),
+    ),
+    VaultTemplate(
+        contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+        "{{ printf .Data.value }}{{ end }}",
+        destination=Path(f"{pomerium_config.certificate_file}"),
+    ),
+]
+consul_configuration = {
+    Path("00-default.json"): ConsulConfig(
+        addresses=ConsulAddresses(dns="127.0.0.1", http="127.0.0.1"),
+        advertise_addr='{{ GetInterfaceIP "ens5" }}',
+        services=[
+            ConsulService(
+                name="airbyte",
+                port=DEFAULT_HTTPS_PORT,
+                check=ConsulServiceTCPCheck(
+                    name="airbyte-https",
+                    tcp="localhost:443",
```

Comment:
Are the other services that discover this node going to be communicating via HTTPS on the local network? If so, will they need a different TLS certificate to support access at a `.consul` TLD?

Reply:
Okay, so expose a :80 listener?

Reply:
Yeah, it's worth allowing Nginx to accept traffic over port 80 for that host